### PR TITLE
Fix : quantstats

### PR DESCRIPTION
**Pull request by prithviraj-gotepatil-algobulls**
- added new parameters name
- conversion of parameters before sanity checks
- accepts many parameters as string input for BT and PT
- added function for token URL
- added live trading function
- added delete previous trades function
- resolved time input for papertrade

### CODE REVIEW

File: pyalgotrading/algobulls/connection.py [hunk 1]
```diff
@@ -247,61 +259,105 @@ def get_report(self, strategy_code, trading_type, report_type, render_as_datafra
...
-            strategy_mode: Intraday or delivery
+            parameters: Parameters
+            candle: Candle interval
+            mode: Intraday or delivery
+
+        Legacy args (will be deprecated in future release):
+            'strategy_code' behaves same 'strategy'
+            'start_timestamp' behaves same 'start'
+            'end_timestamp' behaves same 'end'
+            'instrument' behaves same 'instrumetns'
```
Comment (pushpakdagade) [inline pyalgotrading/algobulls/connection.py:280]: spelling mistake; pls run spellcheck and fix all mistakes.

File: pyalgotrading/algobulls/connection.py [hunk 2]
```diff
@@ -450,61 +509,105 @@ def get_backtesting_report_order_history(self, strategy_code):
         assert isinstance(strategy_code, str), f'Argument "strategy_code" should be a string'
         return self.get_report(strategy_code=strategy_code, trading_type=TradingType.BACKTESTING, report_type=TradingReportType.ORDER_HISTORY)
 
-    def papertrade(self, strategy_code, start_time, end_time, instrument, lots, strategy_parameters, candle_interval, strategy_mode=StrategyMode.INTRADAY):
+    def papertrade(self, strategy=None, start=None, end=None, instruments=None, lots=None, parameters=None, candle=None, strategy_mode=StrategyMode.INTRADAY, **kwargs):
         """
-        Start a paper trading session
+        Submit a backtesting job for a strategy on the AlgoBulls Platform
```
Comment (pushpakdagade) [inline pyalgotrading/algobulls/connection.py:514]: you forgot to replace backtesting with paper trading;

File: pyalgotrading/algobulls/connection.py [hunk 3]
```diff
@@ -450,61 +509,105 @@ def get_backtesting_report_order_history(self, strategy_code):
...
+        mode = mode if 'strategy_mode' not in kwargs else kwargs.get('strategy_mode')
+
+        # Sanity checks - Convert config parameters
+        _error_msg_candle = f'Argument "candle" should be a valid string or an enum of type CandleInterval. Possible string values can be: {get_valid_enum_names(CandleInterval)}'
+        _error_msg_timestamps = f'Argument "start" should be a valid timestamp string (YYYY-MM-DD | HH:MM) or an instance of type datetime.datetime'
+        _error_msg_instruments = f'Argument "instruments" should be a valid instrument string or a list of valid instruments strings. You can use the \'get_instrument()\' method of AlgoBullsConnection class to search for instruments'
+        _error_msg_mode = f'Argument "mode" should be a valid string or an enum of type StrategyMode. Possible string values can be: {get_valid_enum_names(StrategyMode)}'
+
+        if isinstance(start, str):
+            start = dt.strptime(start, '%Y-%m-%d | %H:%M')
```
Comment (pushpakdagade) [inline pyalgotrading/algobulls/connection.py:555]: for pt, we just take time and not datetime. So there are errors below.

File: pyalgotrading/algobulls/connection.py (hunk 1 above)
Comment (prithviraj-gotepatil-algobulls) [reply pyalgotrading/algobulls/connection.py]: done

**Review by pushpakdagade — APPROVED** [review]

**Review by pushpakdagade — APPROVED** [review]

**Pull request merged**